eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/AbstractSynchronizedPutMultimap.java

```diff
@@ -93,7 +95,29 @@ public boolean put(K key, V value)
         }
     }
 
-    private C getIfAbsentPutCollection(K key)
+    @Override
+    public MutableMultimap<K, V> withKeyMultiValues(K key, V... values)
+    {
+        if (ArrayIterate.notEmpty(values))
+        {
+            C existingValues = this.getIfAbsentPutCollection(key);
+            synchronized (existingValues)
+            {
+                this.addAll(existingValues, values);
+                return this;
+            }
+        }
+        return this;
+    }
+
+    protected void addAll(C existingValues, V[] values)
+    {
+        int currentSize = existingValues.size();
+        int newSize = ArrayIterate.addAllTo(values, existingValues).size();
+        this.addToTotalSize(newSize - currentSize);
+    }
+
+    protected C getIfAbsentPutCollection(K key)
```

Comment:
Let's keep this method private, and use the casting approach in subclasses.

Reply:
Addressed.